azure-pipelines/end-to-end-tests-dir/asset-caching.ps1

```diff
@@ -105,3 +103,166 @@ $expectedPattern = "(?s)" +
 if (-not ($actual -match $expectedPattern)) {
     throw "Failure: couldn't find expected debug promises (asset caching disabled + x-block-origin disabled)"
 }
+
+
+# azurl (no), x-block-origin (no), asset-cache (n/a), download (fail)
+# Expected: Download failure message, nothing about asset caching
+Refresh-TestRoot
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a", "--url", "https://localhost:1234/foobar.html"))
+$actual = $actual -replace "`r`n", "`n"
+
+$expected = @(
+"error: https://localhost:1234/foobar.html: curl failed to download with exit code 7"
+) -join "`n"
+
+if (-not ($actual -match $expected)) {
+    throw "Failure: azurl (no), x-block-origin (no), asset-cache (n/a), download (fail)"
+}
+
+#azurl (no), x-block-origin (no), asset-cache (n/a), download (sha-mismatch)
+#Expected: Download message with "you might need to configure a proxy" and expected/actual sha
```

Comment:
Note: No "you might need to configure a proxy". I could add back in, but it feels like a potentially misleading message.

Reply:
Actually, we aren't removing the "you might need to configure a proxy" message from cmake world. https://github.com/microsoft/vcpkg/pull/40945/files#diff-95f7d8e8b25d1928ee2738036384adec841ddd80c51a526f45e9df167fea5b37L31

https://github.com/microsoft/vcpkg/pull/40945/files#diff-95f7d8e8b25d1928ee2738036384adec841ddd80c51a526f45e9df167fea5b37L270